src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -51,66 +51,79 @@ export class ReactNativeErrorHandlers implements Integration {
    */
   private _handleUnhandledRejections(): void {
     if (this._options.onunhandledrejection) {
+      /*
+        In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
+        This is due to a version mismatch between promise versions. The version will need to be fixed with a package resolution.
+        We first run a check and show a warning if needed.
+      */
+      this._checkPromiseVersion();
+
       const tracking: {
         disable: () => void;
         enable: (arg: unknown) => void;
         // eslint-disable-next-line @typescript-eslint/no-var-requires,import/no-extraneous-dependencies
       } = require("promise/setimmediate/rejection-tracking");
 
+      // Default behavior if the React Native promise rejection handlers are not available for some reason
+      let promiseRejectionTrackingOptions = {
+        onUnhandled: (_id: string, error: unknown) => {
+          // eslint-disable-next-line no-console
+          console.warn(error);
+        },
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        onHandled: (_id: string) => {},
+      };
+
+      try {
+        // Here we try to use React Native's original promise rejection handler.
+        // eslint-disable-next-line @typescript-eslint/no-var-requires,@typescript-eslint/no-unsafe-member-access
+        promiseRejectionTrackingOptions = require("react-native/Libraries/promiseRejectionTrackingOptions")
+          .default;
+      } catch (e) {
+        // do nothing, as the user may be running an older version of RN without promiseRejectionTrackingOptions
```

Comment:
Let's please extract setting `promiseRejectionTrackingOptions` into a function and then always return a const. The require check is an implementation detail that `_handleUnhandledRejections` shouldn't have to worry about imo.